// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTStorageNym";
option optimize_for = LITE_RUNTIME;

import public "HDAccount.proto";
import public "StorageBlockchainAccountList.proto";
import public "StoragePurse.proto";
import public "StorageItemHash.proto";

message StorageNym {
    optional uint32 version = 1;
    optional string nymid = 2;
    optional StorageItemHash credList = 3;
    optional StorageItemHash SentPeerRequests = 4;
    optional StorageItemHash IncomingPeerRequests = 5;
    optional StorageItemHash SentPeerReply = 6;
    optional StorageItemHash IncomingPeerReply = 7;
    optional StorageItemHash FinishedPeerRequest = 8;
    optional StorageItemHash FinishedPeerReply = 9;
    optional StorageItemHash ProcessedPeerRequest = 10;
    optional StorageItemHash ProcessedPeerReply = 11;
    optional StorageItemHash MailInbox = 12;
    optional StorageItemHash MailOutbox = 13;
    optional StorageItemHash Threads = 14;
    optional StorageItemHash Contexts = 15;
    optional StorageItemHash Accounts = 16;
    repeated StorageBlockchainAccountList BlockchainAccountIndex = 17;
    repeated HDAccount HDAccount = 18;
    optional string issuers = 19;
    optional string PaymentWorkflow = 20;
    optional string bip47 = 21;
    repeated StoragePurse purse = 22;
}
